vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceController.java

```diff
@@ -72,15 +101,131 @@ public ResponseEntity<List<SpaceLink>> getSpaceLinksPresent(@PathVariable("space
         List<SpaceLink> spaceLinkPresent = spaceManager.getIncomingLinks(spaceId);
         return new ResponseEntity<>(spaceLinkPresent, HttpStatus.OK);
     }
-    
-    @RequestMapping(value = STAFF_SPACE_PATH+"{id}/links", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<Map<String,Object>> showSpaceLinks(@PathVariable String id, Model model) {
-        Map<String,Object> responseData = new HashMap<String,Object>();
+
+    @RequestMapping(value = STAFF_SPACE_PATH
+            + "{id}/links", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Map<String, Object>> showSpaceLinks(@PathVariable String id, Model model) {
+        Map<String, Object> responseData = new HashMap<String, Object>();
+
         responseData.put("spaceLinks", spaceLinkManager.getLinkDisplays(id));
+        
         responseData.put("externalLinks", externalLinkManager.getLinkDisplays(id));
         responseData.put("moduleLinks", moduleLinkManager.getLinkDisplays(id));
         responseData.put("textBlocks", spaceTextBlockManager.getSpaceTextBlockDisplays(id));
         return new ResponseEntity<>(responseData, HttpStatus.OK);
     }
 
+
+    @RequestMapping(value = API_DEFAULT_SPACEIMAGE_PATH, method = RequestMethod.GET)
+    public ResponseEntity<byte[]> getSpaceId() {
```

Comment:
should not be part of this controller; those should be part of their own controller;